repo/modules.go

```diff
@@ -94,17 +95,21 @@ func importRepoRulesModules(filename string, _ *RemoteCache) (repos []Repo, err
 		if mod, ok := pathToModule[path]; ok {
 			mod.Sum = sum
 		}
+		if mod, ok := replacePathToModule[path]; ok {
```

Comment:
Before setting `mod.Sum`, should we also check that the version of the module read from the `go.sum` file matches our module's version?

Reply:
Yes, we should do that. We should skip entries if the versions don't match.